src/HaskellerAnswers.hs

```diff
@@ -43,14 +52,48 @@ data Event
     deriving stock (Eq, Show)
 
 updateApp :: Event -> Model -> Effect Event Model
-updateApp event model = case event of
+updateApp event model@Model{..} = case event of
     NoEvent    -> noEff model
-    NextAnswer -> let cur :| rest = modelAllAnswers model in
-        noEff $ model
-            { modelAllAnswers    = NE.fromList rest
+    NextAnswer ->
+        let (cur, newAnswers, newGen) = pickRandomAnswer modelAllAnswers modelRandGen
+        in noEff $ model
+            { modelAllAnswers    = newAnswers
             , modelCurrentAnswer = cur
+            , modelRandGen       = newGen
             }
 
+{- | This function extracts random element from a given list and returns this
+element and the list without this element. If the given list is empty
+we extract from 'defaultAnswers' instead.
+-}
+pickRandomAnswer :: [Text] -> StdGen -> (Text, [Text], StdGen)
+pickRandomAnswer answers gen =
+    let (pick, newGen) = extractRandom gen answers
```

Comment:
Probably collapse `pickRandomAnswer` and `extractRandom` so reduce the amount of pattern-matching?
Instead of firstly checking on the empty list and then on `Maybe` result, we just can check the list on emptiness and do all other action regarding. What do you think? Could it make it more difficult?

Reply:
@vrom911 Interesting idea :+1: I can try to reduce the number of used functions.